Modules/vc-module-inventory/VirtoCommerce.InventoryModule.Data/Services/InventorySearchService.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+using VirtoCommerce.Domain.Inventory.Model.Search;
+using VirtoCommerce.InventoryModule.Core.Model;
+using VirtoCommerce.InventoryModule.Core.Services;
+using VirtoCommerce.InventoryModule.Data.Cashing;
+using VirtoCommerce.InventoryModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Caching;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace VirtoCommerce.InventoryModule.Data.Services
+{
+    public class InventorySearchService : IInventorySearchService
+    {
+        private readonly Func<IInventoryRepository> _repositoryFactory;
+        private readonly IPlatformMemoryCache _platformMemoryCache;
+        public InventorySearchService(Func<IInventoryRepository> repositoryFactory, IPlatformMemoryCache platformMemoryCache)
+        {
+            _repositoryFactory = repositoryFactory;
+            _platformMemoryCache = platformMemoryCache;
+        }
+
+        public async Task<GenericSearchResult<InventoryInfo>> SearchInventoriesAsync(InventorySearchCriteria criteria)
+        {
+            var cacheKey = CacheKey.With(GetType(), "SearchInventoriesAsync", criteria.GetCacheKey());
```

Comment:
Need to opt of the caching the inventory search due that fact that's is too hard to invalidate these kinds of cache, better need to add caching for the most frequently and perfromance critical used methods such as `GetProductsInventoryInfosAsync` because there is easily to set up a cache invalidation